File: learning-spring-java/src/main/java/com/ironbird/learningspring/data/Reservation.java
package com.ironbird.learningspring.data;

import jakarta.persistence.*;

import java.util.Date;

@Entity
@Table(name = "RESERVATION")
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "RESERVATION_ID")
    private long reservationId;

    @ManyToOne
    @JoinColumn(name = "ROOM_ID", referencedColumnName = "ROOM_ID")
    private Room room;

    @ManyToOne
    @JoinColumn(name = "GUEST_ID", referencedColumnName = "GUEST_ID")
    private Guest guest;

    @Column(name = "RES_DATE")
    private Date reservationDate;

    public long getReservationId() {
        return reservationId;
    }

    public void setReservationId(long id) {
        this.reservationId = id;
    }

    public Room getRoom() {
        return room;
    }

    public void setRoom(Room room) {
        this.room = room;
    }

    public Guest getGuest() {
        return guest;
    }

    public void setGuest(Guest guest) {
        this.guest = guest;
    }

    public Date getReservationDate() {
        return reservationDate;
    }

    public void setReservationDate(Date reservationDate) {
        this.reservationDate = reservationDate;
    }

    @Override
    public String toString() {
        return "Reservation{" +
                "id=" + reservationId +
                ", room=" + room +
                ", guest=" + guest +
                ", reservationDate=" + reservationDate +
                '}';
    }
}

File: learning-spring-kotlin/src/main/kotlin/com/ironbird/learningspring/data/Reservation.kt
package com.ironbird.learningspring.data

import jakarta.persistence.*
import java.util.*

@Entity
@Table(name = "RESERVATION")
class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "RESERVATION_ID")
    var reservationId: Long = 0

    @ManyToOne
    @JoinColumn(name = "ROOM_ID", referencedColumnName = "ROOM_ID")
    var room: Room? = null

    @ManyToOne
    @JoinColumn(name = "GUEST_ID", referencedColumnName = "GUEST_ID")
    var guest: Guest? = null

    @Column(name = "RES_DATE")
    var reservationDate: Date? = null

    override fun toString(): String {
        return "Reservation(reservationId=$reservationId, room=$room, guest=$guest, reservationDate=$reservationDate)"
    }
}
